### DISCUSSION TITLE
OSNet Role and Counting objects

### DISCUSSION THREAD

**Discussion by zohaib-khan5040** [mikel-brostrom/yolo_tracking#424] [Q&A]
Hello,
I trained a model on a dataset with images of cattle using the original YoloV5 repo on Object Detection (the yolov5m model configuration). I'm confused regarding the role of the OSNet model here: what it does and how we go about choosing one for our task (if necessary). Any responses or references to previous discussions will be highly appreciated!

Also, is there any command line argument I could pass while running the tracking script that would also *display the counts of the objects* in the corner of the window? If there isn't, could someone please give a hint on what to tweak to go about doing that?

Thank you!

**Comment by mikel-brostrom** [comment]
> I'm confused regarding the role of the OSNet model here: what it does

It is just an updated model for visual appearance extraction. In order to match people (or any other class you want to track) and distinguish them from impostors, features corresponding small local regions (e.g. shoes, glasses) and global whole body regions are equally important. This is not captured by the simple feature extractor provided by https://github.com/nwojke/deep_sort. For more information on this check: https://arxiv.org/pdf/1905.00953.pdf.

> how we go about choosing one for our task (if necessary).

It depends on what you tracking quality needs are. The default model could be good enough. 

> Also, is there any command line argument I could pass while running the tracking script that would also display the counts of the objects in the corner of the window?

There is no built-in counting functionality

> If there isn't, could someone please give a hint on what to tweak to go about doing that?

https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/issues/428
